internal/pkg/cli/local_run.go

```diff
@@ -0,0 +1,257 @@
+		log.Infof("Only one environment found, defaulting to: %s\n", color.HighlightUserInput(envs[0]))
+		o.envName = envs[0]
+		return nil
+
+	}
+	selectedEnvName, err := o.prompt.SelectOne("Select an environment in which you want to test", "", envs, prompt.WithFinalMessage("Environment:"))
+	if err != nil {
+		return fmt.Errorf("select environment: %w", err)
+	}
+	o.envName = selectedEnvName
+	return nil
+}
+
+func (o *localRunOpts) getDeployedEnvironments() ([]string, error) {
+	envConfig, err := o.store.ListEnvironments(o.appName)
+	if err != nil {
+		return nil, fmt.Errorf("get environments for app %s: %w", o.appName, err)
+	}
+
+	var deployedEnvs []string
+	for _, env := range envConfig {
+		isDeployed, err := o.isEnvironmentDeployed(env.Name)
+		if err != nil {
+			return nil, err
+		}
+		if isDeployed {
+			deployedEnvs = append(deployedEnvs, env.Name)
+		}
+	}
+
+	return deployedEnvs, nil
+}
+
+func (o *localRunOpts) isEnvironmentDeployed(envName string) (bool, error) {
+	var checker versionCompatibilityChecker
+
+	checker, err := o.envVersionChecker(envName)
+	if err != nil {
+		return false, err
+	}
+
+	currVersion, err := checker.Version()
+	if err != nil {
+		return false, fmt.Errorf("get environment %q version: %w", envName, err)
+	}
+	if currVersion == version.EnvTemplateBootstrap {
+		return false, nil
+	}
+	return true, nil
+}
+func (o *localRunOpts) validateEnvName() error {
+	isDeployed, err := o.isEnvironmentDeployed(o.envName)
+	if err != nil {
+		return err
+	}
+	if !isDeployed {
+		return fmt.Errorf(`cannot use an environment which is not deployed Please run "copilot env deploy, --name %s" to deploy the environment first`, o.envName)
+	}
+	return nil
+}
+
+func (o *localRunOpts) validateOrAskWorkloadName() error {
+	if o.wkldName != "" {
+		return o.validateWkldName()
+	}
+	services, err := o.deployStore.ListDeployedServices(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get services: %w", err)
+	}
+
+	jobs, err := o.deployStore.ListDeployedJobs(o.appName, o.envName)
+	if err != nil {
+		return fmt.Errorf("Get jobs: %w", err)
+	}
+
+	workloads := append(services, jobs...)
+	if len(workloads) == 0 {
+		return fmt.Errorf("no workloads found in this environment %s", o.envName)
+	}
+	if len(workloads) == 1 {
+		log.Infof("Only one workload found in this environment, defaulting to: %s\n", color.HighlightUserInput(workloads[0]))
+		o.wkldName = workloads[0]
+	} else {
+		selectedWorloadName, err := o.prompt.SelectOne("Select a workload that you want to run locally", "", workloads, prompt.WithFinalMessage("workload name"))
+		if err != nil {
+			return fmt.Errorf("select Workload: %w", err)
+		}
+		o.wkldName = selectedWorloadName
+	}
+	localWorkloads, err := o.ws.ListWorkloads()
+	if err != nil {
+		return fmt.Errorf("list workloads in the workspace : %w", err)
+	}
+	if !contains(o.wkldName, localWorkloads) {
+		return fmt.Errorf("selected service %q does not exist in the workspace", o.wkldName)
+	}
```

Comment:
In this case, we shouldn't have included these workloads in the options that we prompted with.

Reply:
We need this since  this is a validation when we give svc name as a flag right?

Reply:
When we give svc name as a flag, I thought we'd have hit L180:
```
if o.wkldName != "" {
	return o.validateWkldName()
}
```
and then go straight to `validateWkldName`.

This `if !contains(o.wkldName, localWorkloads)` line will only execute if workload is prompted for 🤔 ?

Reply:
Oh sorry I thought you commented I that functions check!